### PR TITLE
chore(title): title fixes and upgrades

### PR DESCRIPTION
**Pull request by seanforyou23**
**What**: This PR fixes a TS error when trying to set the headingLevel prop for Title. I've gone through the react workspace and updated all instances of `<Title>` and given what I believe is a sensible value for headingLevel, basically corresponding it with the given `size` that was already present. As a result, the demo pages have a true hierarchy that better reflects the structure of the content.

Converts `Title` to a typescript.

Along the way, I noticed `/patternfly-4/react-docs/src/components/docs/` was unused so I deleted it.

With these changes, you'll now get proper IDE support for the level property.

**Additional issues**:
Fixes https://github.com/patternfly/patternfly-react/issues/1378

assign an appropriate headingLevel for each use of <Title> that corresponds to the size prop used
update tests and snapshots
fix a11y issues in workspace
remove unused /patternfly-4/react-docs/src/components/docs/
fix bad path in pagination
convert title to TS

### CODE REVIEW

**Comment by dgutride** [comment]
@seanforyou23  - we can't rename props without approval right now - please open an issue tracking that request and we'd like to get any bug fixes taken in separately

**Comment by seanforyou23** [comment]
No problem @dgutride - https://github.com/patternfly/patternfly-react/issues/1439 captures the request.

**Comment by jschuler** [comment]
@seanforyou23  How about you keep the prop name the same as before and also don't make it required so as not to introduce a breaking change, then this can still go in. We can track prop name change and making it required in a separate issue if need be.

**Comment by seanforyou23** [comment]
Working through those updates now, will push these changes soon

File: packages/patternfly-4/react-docs/src/components/example/example.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ const Example = ({
     const path = `/${pathStart}/examples/${exampleName}`;
     return (
       <Section>
-        <Title size="lg" headingLevel="h3">
+        <Title size="lg" level="h3">
```
Comment (jschuler) [inline packages/patternfly-4/react-docs/src/components/example/example.js]: These ones also need to be changed

**Comment by seanforyou23** [comment]
Alrighty @tlabaj @jschuler @dlabaj - thx for the feedback. I believe it's all addressed in the latest changeset. Let me know if you see anything else!

**Comment by seanforyou23** [comment]
If anyone knows where I can see `liveDemo.js` in action, feel free to point me there and I'll make sure it uses a sensible heading level. Right now it's set to h6 which likely isn't accurate.

File: packages/patternfly-4/react-core/src/components/Title/levels.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const HeadingLevels = {
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Title/levels.js]: Looks like you'll want to add this to the index.js file so it can be imported by consumer?

Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/Title/levels.js]: Is this a user facing component? and is There a need for the separate levels.js file?

**Comment by seanforyou23** [comment]
Sorry for the delay @tlabaj, I was taking a bit of time to brush up on some related TS concepts to help me better understand how to best treat this. The purpose for HeadingLevels was only ever to ensure we have good typing support and intellisense for the headingLevel property. I don't really need or see a reason for it to be a separate module in a separate file, or even consumable by end users, but so far that's the only way I've been able to get it to work.

In working through moving HeadingLevels back into the main Title component file, it came up that this would be a good opportunity to go ahead and do the full conversion to TypeScript so we don't have the manually manage the declaration files. That's pretty much where I'm at. This opened a can of worms that I'm still trying to sort through.

If it's ok I'd like to take a little more time to get the best solution in place and make sure I understand what's at stake myself. Is that ok?

**Comment by seanforyou23** [comment]
Ok, I've addressed quite a few issues and with the exception of the prop table content, Title seems to be working well as a TS component. The `headingLevel` and `size` props don't populate the props table as they did before. I'm still trying to figure out how to fix that. I'd love some feedback on the TitleProps interface in case anyone knows a better way to define size and headingLevel. Thanks for all the help!

**Comment by jschuler** [comment]
@seanforyou23 We have implemented basic props table support so far, as we work on more TS components we will have to add more prop types. It is very possible that one of your props needs yet to be handled correctly to show up in the table.

File: packages/patternfly-4/react-core/src/components/Title/index.d.ts [hunk 3]
```diff
@@ -1 +1,2 @@
-export { default as Title, TitleSize, TitleProps } from './Title';
+export { default as Title, TitleProps } from './Title';
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Title/index.d.ts]: I think you can delete this file

File: packages/patternfly-4/react-styles/src/utils.d.ts [hunk 4]
```diff
@@ -13,7 +14,7 @@ export function getModifier(
   styleObject: any,
   modifier: string,
   defaultModifer?: StyleDeclarationStatic | string
-): StyleDeclarationStatic | string;
+): ClassNameArg | string;
```
Comment (jschuler) [inline packages/patternfly-4/react-styles/src/utils.d.ts]: Revert this change
